src/RelayPricer.ts

```diff
@@ -0,0 +1,28 @@
+import { SourceApi } from './types/RelayPricer';
+
+export default class RelayPricer {
+    private readonly sourceApi: SourceApi;
+
+    constructor(sourceApi: SourceApi) {
+        this.sourceApi = sourceApi;
+    }
+
```

Comment:
What if I pass a null/undefined sourceApi to this constructor?

Reply:
add test for this.

Reply:
updated